src/style/style.js

```diff
@@ -363,24 +363,25 @@ class Style extends Evented {
         }
     }
 
-    _updateProjection() {
-        if (!this.map._explicitProjection) {
+    _updateMapProjection() {
+        if (!this.map._explicitProjection) { // Update the visible projection if map's is null
             this.map._updateProjection();
         } else { // Ensure that style is consistent with current projection on style load
-            this.enableDraping();
+            this.updateProjection();
         }
     }
 
-    enableDraping() {
-        if (this._loaded) {
-            if (this.map.transform.projection.requiresDraping) {
-                const hasTerrain = this.getTerrain() || this.stylesheet.terrain;
-                if (!hasTerrain) {
-                    this.setTerrainForDraping();
-                }
-            } else if (this.terrainSetForDrapingOnly()) {
-                this.setTerrain(null);
+    updateProjection() {
+        if (!this._loaded) return;
+        this.dispatcher.broadcast('setProjection', this.map.transform.projectionOptions);
+
+        if (this.map.transform.projection.requiresDraping) {
+            const hasTerrain = this.getTerrain() || this.stylesheet.terrain;
+            if (!hasTerrain) {
+                this.setTerrainForDraping();
             }
+        } else if (this.terrainSetForDrapingOnly()) {
+            this.setTerrain(null);
         }
     }
```

Comment:
Is there any way we could improve naming to better differente these functions? We have:
1. `style._updateMapProjection` that calls `style.updateProjection` or `map._updateProjection`
2. `style.updateProjection` that dispatches worker events, sets mock draping
3. `map._updateProjection` that manages the transition update, updates the transform's projeciton, eventually calls `style.updateProjection` (Possibly cyclic calls with 1.?)

Reply:
`map._updateProjection` and `style._updateProjection` seems fine, only `style.updateProjection` could be renamed (2.) to better reflect the interaction/event with workers and drape initialization.

Reply:
Good point @karimnaaji! enableDrapingAndNotifyWorkersOfProjectionUpdate` seems a bit long. What do you think of the name `onUpdateProjection` reflecting that these are the updates that need to happen after a projection update?

Reply:
>`enableDrapingAndNotifyWorkersOfProjectionUpdate` seems a bit long

😄 

`onProjectionChanged` is fine, otherwise `style.applyProjectionUpdate`